### PR TITLE
Check if `getElement()` is undefined before adding focus listener (Canvas)

### LINKED ISSUE #8442: Tooltips throw exceptions when using vector shapes with the canvas renderer after #8247
### Checklist

- [X] I've looked at the [documentation](https://leafletjs.com/reference.html) to make sure the behavior isn't documented and expected.
- [X] I'm sure this is an issue with Leaflet, not with my app or other dependencies (Angular, Cordova, React, etc.).
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this hasn't been reported yet.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Steps to reproduce

1. Use latest `main` (9d1a841 at the time of creating this issue)
2. Set up a map with canvas renderer and a vector shape, like a rectangle.
3. Bind a tooltip to the vector. _Check JS console._
4. Move mouse over the vector. _Check JS console._
5. Move mouse out of the vector. _Tooltip does not disappear._

### Expected behavior

Tooltip does not fail to initialise or when reacting to events. They do not require DOM elements to function.

### Current behavior

Exceptions thrown on: tooltip initialisation, mouse events due to `getElement` calls introduced in #8247 671ebdf on a canvas-rendered shape (`getElement` is defined on the class, but has no element to return).

### Minimal example reproducing the issue

https://codepen.io/alex4401/pen/XWqpKMW

### Environment

- Leaflet version: main `9d1a841`
- Browser (with version): Firefox 104.0.2, but got user reports from all other browsers
- OS/Platform (with version): Arch Linux

### PR DESCRIPTION
**Pull request by Falke-Design**
Fix #8442

### CODE REVIEW

File: src/layer/Tooltip.js [hunk 1]
```diff
@@ -387,15 +387,19 @@ Layer.include({
 	},
 
 	_addFocusListenersOnLayer: function (layer) {
-		DomEvent.on(layer.getElement(), 'focus', function () {
-			this._tooltip._source = layer;
-			this.openTooltip();
-		}, this);
-		DomEvent.on(layer.getElement(), 'blur', this.closeTooltip, this);
+		if (layer.getElement()) {
+			DomEvent.on(layer.getElement(), 'focus', function () {
```
Comment (mourner) [inline src/layer/Tooltip.js:391]: Nit: while we're at it, let's avoid repeated calls to the same thing. E.g. do `var el = layer.getElement();` first.

**Review by mourner — APPROVED** [review]

File: src/layer/Tooltip.js [hunk 2]
```diff
@@ -387,15 +387,19 @@ Layer.include({
...
+				this._tooltip._source = layer;
+				this.openTooltip();
+			}, this);
+			DomEvent.on(layer.getElement(), 'blur', this.closeTooltip, this);
+		}
```
Comment (Falke-Design) [inline src/layer/Tooltip.js:396]:
```suggestion
		var el = layer.getElement();
		if (el) {
			DomEvent.on(el, 'focus', function () {
				this._tooltip._source = layer;
				this.openTooltip();
			}, this);
			DomEvent.on(el, 'blur', this.closeTooltip, this);
		}
```

File: src/layer/Tooltip.js [hunk 3]
```diff
@@ -387,15 +387,19 @@ Layer.include({
...
 	},
 
 	_setAriaDescribedByOnLayer: function (layer) {
-		layer.getElement().setAttribute('aria-describedby', this._tooltip._container.id);
+		if (layer.getElement()) {
+			layer.getElement().setAttribute('aria-describedby', this._tooltip._container.id);
+		}
```
Comment (Falke-Design) [inline src/layer/Tooltip.js:402]:
```suggestion
		var el = layer.getElement();
		if (el) {
			el.setAttribute('aria-describedby', this._tooltip._container.id);
		}
```

File: src/layer/Tooltip.js (hunk 1 above)
Comment (Falke-Design) [reply src/layer/Tooltip.js]: Changed

**Pull request merged**